janus_core/calculations/phonons.py

```diff
@@ -0,0 +1,421 @@
+            `file_prefix`.
+        force_const_file : Optional[PathLike]
+            Name of hdf5 file to save force constants. Default is inferred from
+            `file_prefix`.
+        bands_file : Optional[PathLike]
+            Name of yaml file to save band structure. Default is inferred from
+            `file_prefix`.
+        """
+        params_file = params_file if params_file else f"{self.file_prefix}-params.yml"
+        bands_file = bands_file if bands_file else f"{self.file_prefix}-auto_band.yml"
+        if not force_const_file:
+            force_const_file = f"{self.file_prefix}-force_consts.hdf5"
+
+        phonon = self.results["phonon"]
+
+        phonon.save(params_file, settings={"force_constants": False})
+        write_force_constants_to_hdf5(phonon.force_constants, filename=force_const_file)
+        phonon.auto_band_structure(write_yaml=True, filename=bands_file)
+
+    def calc_thermal_props(self, write_results: bool = True) -> None:
+        """
+        Calculate thermal properties and optionally write results.
+
+        Parameters
+        ----------
+        write_results : bool
+            Whether to write out results to file. Default is True.
+        """
+        # Calculate phonons is not already run
+        if "phonon" not in self.results:
+            self.calc_phonons(write_results=False)
+
+        if self.logger:
+            self.logger.info("Beginning thermal properties calculation")
+
+        self.results["phonon"].run_thermal_properties(
+            t_step=self.t_step, t_max=self.t_max, t_min=self.t_min
+        )
+        self.results["thermal_properties"] = self.results[
+            "phonon"
+        ].get_thermal_properties_dict()
+
+        if self.logger:
+            self.logger.info("Thermal properties calculation complete")
+
+        if write_results:
+            self.write_thermal_props()
+
+    def write_thermal_props(self, filename: Optional[PathLike] = None) -> None:
+        """
+        Write results of thermal properties calculations.
+
+        Parameters
+        ----------
+        filename : Optional[PathLike]
+            Name of data file to save thermal properties. Default is inferred from
+            `file_prefix`.
+        """
+        filename = filename if filename else f"{self.file_prefix}-cv.dat"
+
+        with open(filename, "w", encoding="utf8") as out:
+            temps = self.results["thermal_properties"]["temperatures"]
+            c_vs = self.results["thermal_properties"]["heat_capacity"]
+            entropies = self.results["thermal_properties"]["entropy"]
+            free_energies = self.results["thermal_properties"]["free_energy"]
+
+            print("#Temperature [K] | Cv | H | S ", file=out)
+            for temp, c_v, free_energy, entropy in zip(
+                temps, c_vs, free_energies, entropies
+            ):
+                print(f"{temp} {c_v} {free_energy} {entropy}", file=out)
```

Comment:
Default separator of print is a space, since you aren't taking advantage of tablular formatting here, could be better expressed as:
```suggestion
                print(temp, c_v, free_energy, entropy, file=out)
```
or if you just had the `tuple` rather than unpacking it
```suggestion
                print(*line, file=out)
```